### PR TITLE
Fix documentation typos

### PR DESCRIPTION
**Pull request by yurloc**
Fix a few issues discovered while reading the constraint streams chapter.

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>specific pull request build</b> please add comment: <b>Jenkins (re)run [optaplanner|apps|examples] tests</b>
* for a <b>full downstream build</b> please add comment: <b>Jenkins run fdb</b>
* for a <b>compile downstream build</b> please add comment: <b>Jenkins run cdb</b>
* for a <b>full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>
* for an <b>upstream build</b> please add comment: <b>Jenkins run upstream</b>
* for a <b>Quarkus LTS check</b> please add comment: <b>Jenkins run LTS</b>
* for a <b>specific Quarkus LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner|apps|examples] LTS</b>
* for a <b>Native check</b> please add comment: <b>Jenkins run native</b>
* for a <b>specific Native LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner|apps|examples] native</b>
</details>

### CODE REVIEW

**Review by triceo — APPROVED** [review]

File: optaplanner-docs/src/main/asciidoc/ConstraintStreams/ConstraintStreams-chapter.adoc [hunk 1]
```diff
@@ -594,7 +594,7 @@ On `UniConstraintStream`, the `ifExistsOther()` building block is also available
 `from()` constraint match type is the same as the `ifExists()` type.
 
 Conversely, if the `ifNotExists()` building block is used (as well as the `ifNotExistsOther()` building block on
-`UniConstraintStream`) you can achieve the opposite affect:
+`UniConstraintStream`) you can achieve the opposite effect:
```
Comment (yurloc) [inline optaplanner-docs/src/main/asciidoc/ConstraintStreams/ConstraintStreams-chapter.adoc:597]: Not 100% sure but I believe this is a mistake. Affect seems to be exclusively tied to emotions: https://www.merriam-webster.com/dictionary/affect. @triceo please confirm.

File: optaplanner-docs/src/main/asciidoc/ConstraintStreams/ConstraintStreams-chapter.adoc [hunk 2]
```diff
@@ -790,7 +790,7 @@ In this case, the test data includes two instances of the `Queen` planning entit
         Queen queen2 = new Queen(1, row1, column2);
 ----
 
-Further down, the following code test the constraint:
+Further down, the following code tests the constraint:
```
Comment (yurloc) [inline optaplanner-docs/src/main/asciidoc/ConstraintStreams/ConstraintStreams-chapter.adoc:793]: Confirm please.

**Comment by yurloc** [comment]
@triceo Done reading the constraint streams chapter. No more updates on this PR.

**Comment by yurloc** [comment]
I'll use this PR to check whether the FDB jobs was fixed by
- https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1671
- https://github.com/kiegroup/kie-jenkins-scripts/pull/1011

Jenkins run fdb.

File: optaplanner-docs/src/main/asciidoc/ConstraintStreams/ConstraintStreams-chapter.adoc (hunk 1 above)
Comment (triceo) [reply optaplanner-docs/src/main/asciidoc/ConstraintStreams/ConstraintStreams-chapter.adoc:597]: Yes, correct.

File: optaplanner-docs/src/main/asciidoc/ConstraintStreams/ConstraintStreams-chapter.adoc (hunk 2 above)
Comment (triceo) [reply optaplanner-docs/src/main/asciidoc/ConstraintStreams/ConstraintStreams-chapter.adoc:793]: Correct.

**Review by triceo — APPROVED** [review]

**Comment by yurloc** [comment]
[ERROR] Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.7.0.1746:sonar (default) on project optaplanner-examples: Could not find a default branch to fall back on. -> [Help 1]

**Comment by yurloc** [comment]
**Linux - Full Downstream Build**

[ERROR] Failed to execute goal on project jbpm-wb-showcase: Could not resolve dependencies for project org.jbpm:jbpm-wb-showcase:war:7.57.0-SNAPSHOT: Failed to collect dependencies at org.apache.chemistry.opencmis:chemistry-opencmis-commons-impl:jar:0.11.0 -> com.sun.xml.ws:jaxws-rt:jar:2.1.7 -> com.sun.xml.stream.buffer:streambuffer:jar:0.9 -> org.jvnet.staxex:stax-ex:jar:RELEASE: Failed to read artifact descriptor for org.jvnet.staxex:stax-ex:jar:RELEASE: Failed to resolve version for org.jvnet.staxex:stax-ex:jar:RELEASE -> [Help 1]
00:55:16  org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal on project jbpm-wb-showcase: Could not resolve dependencies for project org.jbpm:jbpm-wb-showcase:war:7.57.0-SNAPSHOT: Failed to collect dependencies at org.apache.chemistry.opencmis:chemistry-opencmis-commons-impl:jar:0.11.0 -> com.sun.xml.ws:jaxws-rt:jar:2.1.7 -> com.sun.xml.stream.buffer:streambuffer:jar:0.9 -> org.jvnet.staxex:stax-ex:jar:RELEASE

**Comment by yurloc** [comment]
Jenkins run examples.

**Comment by yurloc** [comment]
Jenkins run examples tests.

**Comment by yurloc** [comment]
**Linux - Pull Request**

[ERROR] Failed to execute goal on project optaplanner-core: Could not resolve dependencies for project org.optaplanner:optaplanner-core:jar:8.9.0-SNAPSHOT: Could not find artifact org.kie.kogito:kogito-legacy-api:jar:2.0.0-SNAPSHOT in mirror-central (https://bxms-qe.rhev-ci-vms.eng.rdu2.redhat.com:8443/nexus/content/groups/kie-all/) -> [Help 1]

**Comment by yurloc** [comment]
Jenkins run fdb.

**Comment by yurloc** [comment]
Jenkins run examples tests.

**Comment by yurloc** [comment]
Jenkins retest this.

**Comment by yurloc** [comment]
Jenkins run fdb.

**Comment by yurloc** [comment]
Jenkins retest this.

**Comment by yurloc** [comment]
Jenkins retest this.

**Comment by yurloc** [comment]
Jenkins run fdb.

**Comment by yurloc** [comment]
Jenkins run fdb.

**Pull request merged**